uber/models.py

```diff
@@ -593,27 +593,12 @@ def auto_badge_num(self, badge_type):
 
                 # Searches badge range for a gap in badge numbers; if none found, returns the latest badge number + 1
                 # Doing this lets admins manually set high badge numbers without filling up the badge type's range.
-                first = 0
-                last = sametype.count() - 1
-                if sametype_list[last] - sametype_list[first] == last:
```

Comment:
This was kinda-sorta the cause of #2020, specifically the `==`.

I already knew that the while loop could hang indefinitely in some cases, because even though it has an `else` statement to return the badge num + 1 it didn't always seem to be hitting it in cases where there were no gaps. I put this if check in to bypass the `while` loop in cases where there were no gaps.

Unfortunately, if you have duplicate numbers, this `if` statement won't get hit and you're at the mercy of the I-pulled-this-off-Stack-Overflow while loop below. Oops.

Luckily, I found a much more sensible way to do all this that means we don't need this check anymore at all, because there is no `while` loop so there are no hangs. I am kind of embarrassed that I didn't think of this the first time.